tests/static_map/shared_memory_test.cu

```diff
@@ -93,31 +98,35 @@ TEMPLATE_TEST_CASE_SIG("Shared memory static map",
 
   // using std::unique_ptr because static_map does not have copy/move constructor/assignment
   // operator yet
-  std::vector<std::unique_ptr<MapType>> maps;
+  std::vector<std::unique_ptr<map_type>> maps;
   for (std::size_t map_id = 0; map_id < number_of_maps; ++map_id) {
-    maps.push_back(std::make_unique<MapType>(
-      map_capacity, cuco::empty_key<Key>{-1}, cuco::empty_value<Value>{-1}));
+    maps.push_back(std::make_unique<map_type>(
+      extent_type{}, cuco::empty_key<Key>{-1}, cuco::empty_value<Value>{-1}));
   }
 
   thrust::device_vector<bool> d_keys_exist(number_of_maps * elements_in_map);
   thrust::device_vector<bool> d_keys_and_values_correct(number_of_maps * elements_in_map);
 
+  using ref_type = decltype(maps.front()->ref(cuco::experimental::op::insert));
```

Comment:
Can we instead use `map_type::ref_type`, i.e., ref without any operators and then in the kernel dispatch the ref with `insert`?

Reply:
```cpp
using ref_type = typename map_type::ref_type<cuco::experimental::op::insert_tag>;
```
Cannot be better. Good call!